EasyPost.Tests/ErrorTest.cs

```diff
@@ -29,10 +36,107 @@ public async Task TestBadParameters()
         }
 
         [Fact(Skip = "Test is no longer valid")]
-        public Task TestEmptyApiKey()
+        public void TestEmptyApiKey()
         {
+            UseVCR("empty_api_key");
+
             // No longer possible to have an empty API key
+        }
+
+        [Fact]
+        public Task TestKnownApiExceptionGeneration()
+        {
+            UseVCR("known_api_exception_generation");
+
+            // all the error status codes the library should be able to handle
+            Dictionary<int, Type> exceptionsMap = new Dictionary<int, Type>
+            {
+                { 0, typeof(VcrError) }, // EasyVCR uses 0 as the status code when a recording cannot be found
+                { 401, typeof(UnauthorizedError) },
+                { 402, typeof(PaymentError) },
+                { 403, typeof(UnauthorizedError) },
+                { 404, typeof(NotFoundError) },
+                { 405, typeof(MethodNotAllowedError) },
+                { 408, typeof(TimeoutError) },
+                { 422, typeof(InvalidRequestError) },
+                { 429, typeof(RateLimitError) },
+                { 500, typeof(InternalServerError) },
+                { 503, typeof(ServiceUnavailableError) },
+                { 504, typeof(GatewayTimeoutError) }
+            };
+            foreach (KeyValuePair<int, Type> exceptionDetails in exceptionsMap)
```

Comment:
Thought: I'd like to be testing each type to ensure they contain the properties/keys we'd expect to pretty please.

Reply:
Resolved